src/lib/hooks/use-best-quote/utils/issuance/legacy-quote.ts

```diff
@@ -0,0 +1,181 @@
+    )
+    const [components, units] =
+      await debtIssuanceProvider.getComponentRedemptionUnits(
+        inputToken.address! as Address,
+        inputTokenAmount,
+      )
+
+    const isIcReth = isSameAddress(
+      inputToken.address!,
+      LeveragedRethStakingYield.address!,
+    )
+    const outputTokens = isIcReth
+      ? [RETH]
+      : components.map((component) => {
+          const outputToken = getTokenDataByAddress(component, chainId)!
+          return {
+            ...outputToken,
+            // all these properties below will be irrelevant for the legacy redemption
+            image: outputToken.logoURI,
+            coingeckoId: '',
+            fees: { streamingFee: '0' },
+            indexTypes: [],
+            isDangerous: false,
+            url: '',
+          }
+        })
+
+    const inputTokenPrice = await getTokenPrice(inputToken, chainId)
+    const outputTokenPricesPromises = outputTokens.map((outputToken) => {
+      return getTokenPrice(outputToken, chainId)
+    })
+    const outputTokenPrices = await Promise.all(outputTokenPricesPromises)
+    const outputTokenPricesUsd = outputTokenPrices.map(
+      (price, index) =>
+        price * Number(formatWei(units[index], outputTokens[index].decimals)),
+    )
+    const outputTokenAmountUsd = outputTokenPricesUsd.reduce((total, price) => {
+      return total + price
+    }, 0)
+
+    const callData = encodeFunctionData({
+      abi: DebtIssuanceModuleV2Abi,
+      functionName: 'redeem',
+      args: [
+        inputToken.address! as Address,
+        inputTokenAmount,
+        account as Address,
+      ],
+    })
+
+    const transaction: QuoteTransaction = {
+      account,
+      chainId: 1,
+      from: account,
+      to: contract,
+      data: callData,
+      value: undefined,
+    }
+
+    const defaultGas = 200_000
+    const defaultGasEstimate = BigInt(defaultGas)
+    const gasEstimatooor = new GasEstimatooor(publicClient, defaultGasEstimate)
+    const canFail = false
+    const gasEstimate = await gasEstimatooor.estimate(transaction, canFail)
+    const gasCosts = gasEstimate * gasPrice
+    const gasCostsInUsd = getGasCostsInUsd(gasCosts, nativeTokenPrice)
+    transaction.gasLimit = BigNumber.from(gasEstimate.toString())
+
+    const inputTokenAmountUsd =
+      parseFloat(formatWei(inputTokenAmount, inputToken.decimals)) *
+      inputTokenPrice
+    const outputTokenAmountUsdAfterFees = outputTokenAmountUsd - gasCostsInUsd
+
+    const fullCostsInUsd = getFullCostsInUsd(
+      inputTokenAmount,
+      gasEstimate * gasPrice,
+      inputToken.decimals,
+      inputTokenPrice,
+      nativeTokenPrice,
+    )
+
+    const isMultiComponentRedemption = components.length > 1 && !isIcReth
+    const outputTokenAmount = isMultiComponentRedemption ? BigInt(0) : units[0]
+
+    return {
+      extended: {
+        components: [...components],
+        outputTokens,
+        outputTokenPricesUsd,
+        units: [...units],
+      },
```

Comment:
As mentioned before extending the standardized quote format to allow for the custom output in the case of legacy ui/quotes